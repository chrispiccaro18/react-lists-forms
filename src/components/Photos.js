import React from 'react';
import PropTypes from 'prop-types';
import Photo from './Photo';

function Photos({ photos }) {
  const photosLis = photos.map(photo => (
    <li key={photo}>
      <Photo photo={photo} />
    </li>
  ));

  return (
    <ul>
      {photosLis}
    </ul>
  );
}

Photos.propTypes = {
  photos: PropTypes.array.isRequired
};

export default Photos;
