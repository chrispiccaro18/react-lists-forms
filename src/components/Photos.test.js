import React from 'react';
import { shallow } from 'enzyme';
import Photos from './Photos';

describe('Photos component', () => {
  it('renders a Photos', () => {
    const wrapper = shallow(<Photos photos={['img-string0', 'img-string1', 'img-string2']} />);
    expect(wrapper).toMatchSnapshot();
  });
});

