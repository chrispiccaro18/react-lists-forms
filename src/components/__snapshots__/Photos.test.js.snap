// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos component renders a Photos 1`] = `
<ul>
  <li
    key="img-string0"
  >
    <Photo
      photo="img-string0"
    />
  </li>
  <li
    key="img-string1"
  >
    <Photo
      photo="img-string1"
    />
  </li>
  <li
    key="img-string2"
  >
    <Photo
      photo="img-string2"
    />
  </li>
</ul>
`;
